libsolidity/parsing/Parser.cpp

```diff
@@ -1981,6 +1986,26 @@ pair<vector<ASTPointer<Expression>>, vector<ASTPointer<ASTString>>> Parser::pars
 	return ret;
 }
 
+optional<string> Parser::findLicenseString()
+{
+	// We circumvent the scanner here, because it skips non-docstring comments.
+	static regex const licenseRegex("SPDX-License-Identifier:\\s+([^\\n\\r]+)");
```

Comment:
As it is now, this can also be found in strings. Do we want to extend this to `^// SPDX...`? Then it would not work for `/*`-comments, though.